nightfall-optimist/src/event-handlers/block-proposed.mjs

```diff
@@ -78,35 +80,23 @@ async function blockProposedEventHandler(data) {
             blockNumberL2: block.blockNumberL2,
             mempool: false,
             offchain: true,
+            fromBlockProposer: true,
             ...tx,
           }); // must be offchain or we'll have seen them, mempool = false
         }
         return true;
       }),
     );
 
-    // Update the nullifiers we have stored, with the blockhash. These will
-    // be deleted if the block check fails and we get a rollback.  We do this
-    // before running the block check because we want to delete the nullifiers
-    // asociated with a failed block, and we can't do that if we haven't
-    // associated them with a blockHash.
-    await stampNullifiers(
-      transactions
-        .map(tx => [
-          ...tx.nullifiers.filter(nulls => nulls !== ZERO),
-          ...tx.nullifiersFee.filter(nulls => nulls !== ZERO),
-        ])
-        .flat(Infinity),
-      block.blockHash,
-    );
     // mark transactions so that they are out of the mempool,
     // so we don't try to use them in a block which we're proposing.
-    await removeTransactionsFromMemPool(block.transactionHashes, block.blockNumberL2, timeBlockL2); // TODO is await needed?
+    await removeTransactionsFromMemPool(block.transactionHashes, block.blockNumberL2); // TODO is await needed?
```

Comment:
Why are you removing timeBlockL2?

Reply:
I must have missed this out during rebase

Reply:
timeBlockL2 added back